packages/commonwealth/client/scripts/controllers/app/login.ts

```diff
@@ -178,15 +187,36 @@ export async function updateActiveAddresses({
   } else if (app.user.activeAccounts.length === 0) {
     // no addresses - preview the community
   } else {
-    const existingAddress = app.roles.getDefaultAddressInCommunity({
```

Comment:
Yep that is the plan. Ideally we remove this RolesController entirely. The issue is that it touches lots of places, and its methods have many side effects. So its not such an easy removal.

Reply:
To avoid breaking anything in other places and since it falls out of scope for this PR, I will prefer to not touch `RolesController` or `getDefaultAddressInCommunity` unnecessarily right now. Resolving this convo.